agent/src/main/java/com/teamscale/jacoco/agent/store/upload/teamscale/TeamscaleUploader.java

```diff
@@ -4,46 +4,43 @@
 import com.teamscale.client.ITeamscaleService;
 import com.teamscale.client.TeamscaleServer;
 import com.teamscale.client.TeamscaleServiceGenerator;
-import com.teamscale.jacoco.agent.store.IXmlStore;
-import com.teamscale.jacoco.agent.store.TimestampedFileStore;
+import com.teamscale.jacoco.agent.store.IUploader;
 import com.teamscale.jacoco.agent.util.Benchmark;
 import com.teamscale.jacoco.agent.util.LoggingUtils;
 import okhttp3.MultipartBody;
 import okhttp3.RequestBody;
 import org.slf4j.Logger;
 
+import java.io.File;
 import java.io.IOException;
 
 /** Uploads XML Coverage to a Teamscale instance. */
-public class TeamscaleUploadStore implements IXmlStore {
+public class TeamscaleUploader implements IUploader {
 
 	/** The logger. */
 	private final Logger logger = LoggingUtils.getLogger(this);
 
-	/** The store to write failed uploads to. */
-	private final TimestampedFileStore failureStore;
-
 	/** Teamscale server details. */
 	private final TeamscaleServer teamscaleServer;
 
 	/** Constructor. */
-	public TeamscaleUploadStore(TimestampedFileStore failureStore, TeamscaleServer teamscaleServer) {
-		this.failureStore = failureStore;
+	public TeamscaleUploader(TeamscaleServer teamscaleServer) {
 		this.teamscaleServer = teamscaleServer;
 	}
 
 	@Override
-	public void store(String xml) {
+	public void upload(File coverageFile) {
 		try (Benchmark benchmark = new Benchmark("Uploading report to Teamscale")) {
-			if (!tryUploading(xml)) {
-				logger.warn("Storing failed upload in {}", failureStore.getOutputDirectory());
-				failureStore.store(xml);
+			if (!tryUploading(coverageFile)) {
+				logger.warn("Filed to upload coverage to Teamscale. Won't delete local file");
```

Comment:
typo: "Filed" -> "Failed"
also: please let the user know which file you won't delete and what they can do with it (upload manually)